AustinHarris.JsonRpcTestN/Test.cs

```diff
@@ -1879,6 +1879,13 @@ public void TestNestedReturnType()
             Assert.AreEqual(expected, result.Result);
         }
 
+        [Test]
+        public void InstantiatingAServiceMoreThanOnceDoesNotThrow()
+        {
+            Assert.DoesNotThrow(()=>new CalculatorService());
+            Assert.DoesNotThrow(() => new CalculatorService());
```

Comment:
I am concerned that by allowing this behavior that we would be introducing risk that a service does not behave as expected. This this example we have created two different instances of CalculatorService which may contain different internal state. The way this is being done, currently only the first service will be used to handle requests.
Alternatively a context can be added when registering a service to allow multiple instances to handle requests, each for their own context.